mongoengine/queryset/base.py

```diff
@@ -345,7 +350,7 @@ def insert(
                 insert_func = collection.insert_one
 
         try:
-            inserted_result = insert_func(raw)
+            inserted_result = insert_func(raw, **kwargs)
             ids = (
                 [inserted_result.inserted_id]
                 if return_one
```

Comment:
I see a potential bug related to the following code (that is a bit below in this method):
```
        # Apply inserted_ids to documents
        for doc, doc_id in zip(docs, ids):
            doc.pk = doc_id
```
In case you provide ordered=False and it raises an error (e.g if you try to insert 2 times a document with same pk), that piece of code won't be reached, leaving the mongoengine documents without id's. Instead of relying on `zip` to get the corresponding id, we could rely on the `raw` dict that actually gets updated with the id by pymongo (see this [post](https://stackoverflow.com/q/35191042/6203472)). Now since it will raise an error anyway, perhaps it could be acceptable to not set the id on the Documents instances. What do you think?

Reply:
btw in case of successful unordered insert, I'm also unsure if we could rely on zip when we set `ordered=False`, as I don't know if we can rely on the order of `inserted_result.inserted_ids`